src/catchup/ApplyBucketsWork.cpp

```diff
@@ -208,8 +290,27 @@ ApplyBucketsWork::doWork()
 {
     ZoneScoped;
 
-    // Step 1: apply buckets. Step 2: assume state
-    if (!mSpawnedAssumeStateWork)
+    // Step 1: index buckets. Step 2: apply buckets. Step 3: assume state
+    bool isUsingBucketListDB = mApp.getConfig().isUsingBucketListDB();
+    if (isUsingBucketListDB)
```

Comment:
nit: I'd prefer if we could write this flow such that it's consistent with Work usage across the codebase:
```
if (!mIndexBucketsWork)
{
     // Spawn indexing work for the first time
     mIndexBucketsWork = addWork<IndexBucketsWork>(mBucketsToIndex);
     return State::WORK_RUNNING;
}
else if (mIndexBucketsWork->getState() != BasicWork::State::WORK_SUCCESS)
{
      // Exit early if indexing work is still running, or failed
      return mIndexBucketsWork->getState();
}

// Otherwise, continue with next steps
```